### PR TITLE
Updated storage library versions post release

### PR DESCRIPTION
**Pull request by gapra-msft**

### CODE REVIEW

File: sdk/storage/azure-storage-blob-nio/pom.xml [hunk 1]
```diff
@@ -55,7 +55,7 @@
     <dependency>
       <groupId>com.azure</groupId>
       <artifactId>azure-storage-blob</artifactId>
-      <version>12.2.0</version> <!-- {x-version-update;com.azure:azure-storage-blob;dependency} -->
+      <version>12.3.0</version> <!-- {x-version-update;com.azure:azure-storage-blob;dependency} -->
```
Comment (JimSuplizio) [inline sdk/storage/azure-storage-blob-nio/pom.xml]: This should be **current**, not **dependency**. Libraries built within the same pipeline should have dependencies on the current version of other libraries within that pipeline. This appears to be the only discrepancy in storage.

Comment (gapra-msft) [reply sdk/storage/azure-storage-blob-nio/pom.xml]: @JimSuplizio Just changed it

**Pull request merged**